In [1]:
import pandas as pd

In [2]:
df1 = pd.read_csv('file1.csv')
df2 = pd.read_csv('file2.csv')
df3 = pd.read_csv('file3.csv')


In [3]:
df2.head(5)


Customer          ST GENDER Education Customer Lifetime Value  Income  \
0  GS98873     Arizona      F  Bachelor              323912.47%   16061   
1  CW49887  California      F    Master              462680.11%   79487   
2  MY31220  California      F   College              899704.02%   54230   
3  UH35128      Oregon      F   College             2580706.30%   71210   
4  WH52799     Arizona      F   College              380812.21%   94903   

   Monthly Premium Auto Number of Open Complaints  Total Claim Amount  \
0                    88                    1/0/00               633.6   
1                   114                    1/0/00               547.2   
2                   112                    1/0/00               537.6   
3                   214                    1/1/00              1027.2   
4                    94                    1/0/00               451.2   

      Policy Type  Vehicle Class  
0   Personal Auto  Four-Door Car  
1    Special Auto            SUV  
2   Personal Auto   Two-Door Car  
3   Personal Auto     Luxury Car  
4  Corporate Auto   Two-Door Car

In [4]:
print(df1.shape)
print(df2.shape)
print(df3.shape)

(4008, 11)
(996, 11)
(7070, 11)


In [5]:
print(df1.columns)
print(df2.columns)
print(df3.columns)

Index(['Customer', 'ST', 'GENDER', 'Education', 'Customer Lifetime Value',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Policy Type', 'Vehicle Class', 'Total Claim Amount'],
      dtype='object')
Index(['Customer', 'ST', 'GENDER', 'Education', 'Customer Lifetime Value',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Total Claim Amount', 'Policy Type', 'Vehicle Class'],
      dtype='object')
Index(['Customer', 'State', 'Customer Lifetime Value', 'Education', 'Gender',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Policy Type', 'Total Claim Amount', 'Vehicle Class'],
      dtype='object')


In [6]:
column_names=('CUST','STATE','GENDER','EDUCATION','CUST_LIFE_VALUE','INCOME','MONTHLY_PREM_AUTO','NBR_OPEN_COMPLAINTS','POLICY_TYPE','VEH_CLASS','TOT_AMT_CLAIM')

In [7]:
df1ordered=df1[['Customer', 'ST', 'GENDER', 'Education', 'Customer Lifetime Value',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Policy Type', 'Vehicle Class', 'Total Claim Amount']]
df2ordered=df2[['Customer', 'ST', 'GENDER', 'Education', 'Customer Lifetime Value',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Policy Type', 'Vehicle Class', 'Total Claim Amount']]
df3ordered=df3[['Customer', 'State', 'Gender', 'Education', 'Customer Lifetime Value',
       'Income', 'Monthly Premium Auto', 'Number of Open Complaints',
       'Policy Type', 'Vehicle Class', 'Total Claim Amount']]

In [8]:
df1ordered.columns = column_names
df2ordered.columns = column_names
df3ordered.columns = column_names

In [9]:
df1ordered.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4008 entries, 0 to 4007
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   CUST                 1071 non-null   object 
 1   STATE                1071 non-null   object 
 2   GENDER               954 non-null    object 
 3   EDUCATION            1071 non-null   object 
 4   CUST_LIFE_VALUE      1068 non-null   object 
 5   INCOME               1071 non-null   float64
 6   MONTHLY_PREM_AUTO    1071 non-null   float64
 7   NBR_OPEN_COMPLAINTS  1071 non-null   object 
 8   POLICY_TYPE          1071 non-null   object 
 9   VEH_CLASS            1071 non-null   object 
 10  TOT_AMT_CLAIM        1071 non-null   float64
dtypes: float64(3), object(8)
memory usage: 344.6+ KB


In [10]:
df2ordered.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 996 entries, 0 to 995
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   CUST                 996 non-null    object 
 1   STATE                996 non-null    object 
 2   GENDER               991 non-null    object 
 3   EDUCATION            996 non-null    object 
 4   CUST_LIFE_VALUE      992 non-null    object 
 5   INCOME               996 non-null    int64  
 6   MONTHLY_PREM_AUTO    996 non-null    int64  
 7   NBR_OPEN_COMPLAINTS  996 non-null    object 
 8   POLICY_TYPE          996 non-null    object 
 9   VEH_CLASS            996 non-null    object 
 10  TOT_AMT_CLAIM        996 non-null    float64
dtypes: float64(1), int64(2), object(8)
memory usage: 85.7+ KB


In [11]:
df3ordered.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7070 entries, 0 to 7069
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   CUST                 7070 non-null   object 
 1   STATE                7070 non-null   object 
 2   GENDER               7070 non-null   object 
 3   EDUCATION            7070 non-null   object 
 4   CUST_LIFE_VALUE      7070 non-null   float64
 5   INCOME               7070 non-null   int64  
 6   MONTHLY_PREM_AUTO    7070 non-null   int64  
 7   NBR_OPEN_COMPLAINTS  7070 non-null   int64  
 8   POLICY_TYPE          7070 non-null   object 
 9   VEH_CLASS            7070 non-null   object 
 10  TOT_AMT_CLAIM        7070 non-null   float64
dtypes: float64(2), int64(3), object(6)
memory usage: 607.7+ KB


In [32]:
df_tot.columns

Index(['CUST', 'STATE', 'GENDER', 'EDUCATION', 'CUST_LIFE_VALUE', 'INCOME',
       'MONTHLY_PREM_AUTO', 'NBR_OPEN_COMPLAINTS', 'POLICY_TYPE', 'VEH_CLASS',
       'TOT_AMT_CLAIM'],
      dtype='object')

In [36]:
#Concatenate the datasets
df_tot=pd.concat([df1ordered,df2ordered,df3ordered])
df_tot = df_tot.reset_index()
df_tot=df_tot.drop(columns=['index'])
df_tot

CUST       STATE GENDER             EDUCATION CUST_LIFE_VALUE  \
0      RB50392  Washington    NaN                Master             NaN   
1      QZ44356     Arizona      F              Bachelor      697953.59%   
2      AI49188      Nevada      F              Bachelor     1288743.17%   
3      WW63253  California      M              Bachelor      764586.18%   
4      GA49547  Washington      M  High School or Below      536307.65%   
...        ...         ...    ...                   ...             ...   
12069  LA72316  California      M              Bachelor     23405.98798   
12070  PK87824  California      F               College     3096.511217   
12071  TD14365  California      M              Bachelor     8163.890428   
12072  UP19263  California      M               College     7524.442436   
12073  Y167826  California      M               College     2611.836866   

        INCOME  MONTHLY_PREM_AUTO NBR_OPEN_COMPLAINTS     POLICY_TYPE  \
0          0.0             1000.0              1/0/00   Personal Auto   
1          0.0               94.0              1/0/00   Personal Auto   
2      48767.0              108.0              1/0/00   Personal Auto   
3          0.0              106.0              1/0/00  Corporate Auto   
4      36357.0               68.0              1/0/00   Personal Auto   
...        ...                ...                 ...             ...   
12069  71941.0               73.0                   0   Personal Auto   
12070  21604.0               79.0                   0  Corporate Auto   
12071      0.0               85.0                   3  Corporate Auto   
12072  21941.0               96.0                   0   Personal Auto   
12073      0.0               77.0                   0  Corporate Auto   

           VEH_CLASS  TOT_AMT_CLAIM  
0      Four-Door Car       2.704934  
1      Four-Door Car    1131.464935  
2       Two-Door Car     566.472247  
3                SUV     529.881344  
4      Four-Door Car      17.269323  
...              ...            ...  
12069  Four-Door Car     198.234764  
12070  Four-Door Car     379.200000  
12071  Four-Door Car     790.784983  
12072  Four-Door Car     691.200000  
12073   Two-Door Car     369.600000  

[12074 rows x 11 columns]

In [37]:
df_tot.head(5)

CUST       STATE GENDER             EDUCATION CUST_LIFE_VALUE   INCOME  \
0  RB50392  Washington    NaN                Master             NaN      0.0   
1  QZ44356     Arizona      F              Bachelor      697953.59%      0.0   
2  AI49188      Nevada      F              Bachelor     1288743.17%  48767.0   
3  WW63253  California      M              Bachelor      764586.18%      0.0   
4  GA49547  Washington      M  High School or Below      536307.65%  36357.0   

   MONTHLY_PREM_AUTO NBR_OPEN_COMPLAINTS     POLICY_TYPE      VEH_CLASS  \
0             1000.0              1/0/00   Personal Auto  Four-Door Car   
1               94.0              1/0/00   Personal Auto  Four-Door Car   
2              108.0              1/0/00   Personal Auto   Two-Door Car   
3              106.0              1/0/00  Corporate Auto            SUV   
4               68.0              1/0/00   Personal Auto  Four-Door Car   

   TOT_AMT_CLAIM  
0       2.704934  
1    1131.464935  
2     566.472247  
3     529.881344  
4      17.269323

In [38]:
df_tot.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12074 entries, 0 to 12073
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   CUST                 9137 non-null   object 
 1   STATE                9137 non-null   object 
 2   GENDER               9015 non-null   object 
 3   EDUCATION            9137 non-null   object 
 4   CUST_LIFE_VALUE      9130 non-null   object 
 5   INCOME               9137 non-null   float64
 6   MONTHLY_PREM_AUTO    9137 non-null   float64
 7   NBR_OPEN_COMPLAINTS  9137 non-null   object 
 8   POLICY_TYPE          9137 non-null   object 
 9   VEH_CLASS            9137 non-null   object 
 10  TOT_AMT_CLAIM        9137 non-null   float64
dtypes: float64(3), object(8)
memory usage: 1.0+ MB


In [39]:
df_tot.shape

(12074, 11)

# Here are if variable is categorical or numerical
 0   CUST                 9137 non-null   object THIS IS A CATEGORICAL VALUE. KEY FOR CUSTOMERS
 1   STATE                9137 non-null   object THIS IS A CATEGORICAL VALUE
 2   GENDER               9015 non-null   object THIS IS A CATEGORICAL VALUE
 3   EDUCATION            9137 non-null   object THIS IS A CATEGORICAL VALUE
 4   CUST_LIFE_VALUE      9130 non-null   object THIS SHOULD BE NUMERICAL VALUE
 5   INCOME               9137 non-null   float64THIS IS A NUMERICAL VALUE
 6   MONTHLY_PREM_AUTO    9137 non-null   float64THIS IS A NUMERICAL VALUE
 7   NBR_OPEN_COMPLAINTS  9137 non-null   object THIS SHOULD BE A NUMERICAL VALUE
 8   POLICY_TYPE          9137 non-null   object THIS IS A CATEGORICAL VALUE
 9   VEH_CLASS            9137 non-null   object THIS IS A CATEGORICAL VALUE
 10  TOT_AMT_CLAIM        9137 non-null   float64THIS IS A NUMERICAL VALUE

In [82]:
#Drop columns
df_clean=df_tot.drop(columns=['NBR_OPEN_COMPLAINTS','EDUCATION'])

In [83]:
df_clean["CUST_LIFE_VALUE"]

0                NaN
1         697953.59%
2        1288743.17%
3         764586.18%
4         536307.65%
            ...     
12069    23405.98798
12070    3096.511217
12071    8163.890428
12072    7524.442436
12073    2611.836866
Name: CUST_LIFE_VALUE, Length: 12074, dtype: object

In [84]:
#df_clean['CUST_LIFE_VALUE'] = df_clean['CUST_LIFE_VALUE'].str.strip('%').astype(float)

In [85]:
#df_clean['CUST_LIFE_VALUE']=df_clean['CUST_LIFE_VALUE']*100

In [86]:
#Clean the CUST_LIFE_VALUE variable
df_clean.CUST_LIFE_VALUE=df_clean.CUST_LIFE_VALUE.replace('%','',regex=True).astype(float)*100

In [87]:
df_clean.head(100)

CUST       STATE GENDER  CUST_LIFE_VALUE   INCOME  MONTHLY_PREM_AUTO  \
0   RB50392  Washington    NaN              NaN      0.0             1000.0   
1   QZ44356     Arizona      F       69795359.0      0.0               94.0   
2   AI49188      Nevada      F      128874317.0  48767.0              108.0   
3   WW63253  California      M       76458618.0      0.0              106.0   
4   GA49547  Washington      M       53630765.0  36357.0               68.0   
..      ...         ...    ...              ...      ...                ...   
95  TV87155      Oregon    NaN      209461925.0  69738.0               74.0   
96  KH48895          AZ    NaN       83753539.0  17780.0              109.0   
97  NZ30757  California    NaN       48016615.0  18107.0               62.0   
98  RI22468          AZ    NaN       57459433.0  57740.0               74.0   
99  FZ30935      Oregon    NaN       60661160.0  32627.0               76.0   

       POLICY_TYPE      VEH_CLASS  TOT_AMT_CLAIM  
0    Personal Auto  Four-Door Car       2.704934  
1    Personal Auto  Four-Door Car    1131.464935  
2    Personal Auto   Two-Door Car     566.472247  
3   Corporate Auto            SUV     529.881344  
4    Personal Auto  Four-Door Car      17.269323  
..             ...            ...            ...  
95   Personal Auto  Four-Door Car     492.127532  
96   Personal Auto            SUV     132.588288  
97   Personal Auto  Four-Door Car     297.600000  
98   Personal Auto  Four-Door Car     269.905129  
99   Personal Auto   Two-Door Car     380.036697  

[100 rows x 9 columns]

In [88]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12074 entries, 0 to 12073
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   CUST               9137 non-null   object 
 1   STATE              9137 non-null   object 
 2   GENDER             9015 non-null   object 
 3   CUST_LIFE_VALUE    9130 non-null   float64
 4   INCOME             9137 non-null   float64
 5   MONTHLY_PREM_AUTO  9137 non-null   float64
 6   POLICY_TYPE        9137 non-null   object 
 7   VEH_CLASS          9137 non-null   object 
 8   TOT_AMT_CLAIM      9137 non-null   float64
dtypes: float64(4), object(5)
memory usage: 849.1+ KB


In [106]:
df_clean.value_counts()

CUST     STATE       GENDER  CUST_LIFE_VALUE  INCOME   MONTHLY_PREM_AUTO  POLICY_TYPE     VEH_CLASS      TOT_AMT_CLAIM
MY31220  California  F       8.997040e+07     54230.0  112.0              Personal Auto   Two-Door Car   537.600000       2
CW49887  California  F       4.626801e+07     79487.0  114.0              Special Auto    SUV            547.200000       2
GS98873  Arizona     F       3.239125e+07     16061.0  88.0               Personal Auto   Four-Door Car  633.600000       2
RK45396  California  M       5.380316e+05     0.0      73.0               Corporate Auto  Four-Door Car  350.400000       1
RJ79502  AZ          M       8.501280e+07     98003.0  107.0              Personal Auto   SUV            449.408637       1
                                                                                                                         ..
IS12075  California  F       5.100088e+05     0.0      71.0               Corporate Auto  Four-Door Car  653.726919       1
IR78670  Cali

In [104]:
#Remove records with income <=0
df_filtered=df_clean[df_clean['INCOME']>0]

In [105]:
df_filtered.head(5)

CUST       STATE GENDER  CUST_LIFE_VALUE   INCOME  MONTHLY_PREM_AUTO  \
2  AI49188      Nevada      F      128874317.0  48767.0              108.0   
4  GA49547  Washington      M       53630765.0  36357.0               68.0   
5  OC83172      Oregon      F       82562978.0  62902.0               69.0   
6  XZ87318      Oregon      F       53808986.0  55350.0               67.0   
8  DY87989      Oregon      M      241275040.0  14072.0               71.0   

      POLICY_TYPE      VEH_CLASS  TOT_AMT_CLAIM  
2   Personal Auto   Two-Door Car     566.472247  
4   Personal Auto  Four-Door Car      17.269323  
5   Personal Auto   Two-Door Car     159.383042  
6  Corporate Auto  Four-Door Car     321.600000  
8  Corporate Auto  Four-Door Car     511.200000